rest/admin_api_test.go

```diff
@@ -801,59 +801,51 @@ func TestSessionTtlGreaterThan30Days(t *testing.T) {
 }
 
 func TestSessionExtension(t *testing.T) {
-
-	if testing.Short() {
-		t.Skip("skipping test in short mode")
-	}
-
 	rt := NewRestTester(t, nil)
 	defer rt.Close()
 
 	a := auth.NewAuthenticator(rt.Bucket(), nil)
-	user, err := a.GetUser("")
-	assert.NoError(t, err)
-	user.SetDisabled(true)
-	err = a.Save(user)
-	assert.NoError(t, err)
-
-	user, err = a.GetUser("")
-	assert.NoError(t, err)
-	goassert.True(t, user.Disabled())
-
-	response := rt.SendRequest("PUT", "/db/doc", `{"hi": "there"}`)
-	assertStatus(t, response, 401)
-
-	user, err = a.NewUser("pupshaw", "letmein", channels.SetOf(t, "*"))
+	user, err := a.NewUser("pupshaw", "letmein", channels.SetOf(t, "*"))
 	a.Save(user)
 
-	assertStatus(t, rt.SendAdminRequest("GET", "/db/_session", ""), 200)
+	if err != nil {
+		log.Printf(err.Error())
+	}
 
-	response = rt.SendAdminRequest("POST", "/db/_session", `{"name":"pupshaw", "ttl":1}`)
-	assertStatus(t, response, 200)
+	// More than 10% of expiry (24h)
+	fakeSession := auth.LoginSession{
+		ID:         base.GenerateRandomSecret(),
+		Username:   "user1",
+		Expiration: time.Now().Add(time.Hour * 4),
+		Ttl:        time.Hour * 24,
+	}
 
-	var body db.Body
-	base.JSONUnmarshal(response.Body.Bytes(), &body)
-	sessionId := body["session_id"].(string)
-	sessionExpiration := body["expires"].(string)
-	assert.NotEmpty(t, sessionId)
-	assert.NotEmpty(t, sessionExpiration)
-	assert.True(t, body["cookie_name"].(string) == "SyncGatewaySession")
+	rt.Bucket().Set(docIDForSession(fakeSession.ID), 0, fakeSession)
 
 	reqHeaders := map[string]string{
-		"Cookie": "SyncGatewaySession=" + body["session_id"].(string),
+		"Cookie": "SyncGatewaySession=" + fakeSession.ID,
 	}
-	response = rt.SendRequestWithHeaders("PUT", "/db/doc1", `{"hi": "there"}`, reqHeaders)
-	assertStatus(t, response, 201)
+	response := rt.SendRequestWithHeaders("PUT", "/db/doc1", `{"hi": "there"}`, reqHeaders)
+	log.Printf("PUT Request: Set-Cookie: %v", response.Header().Get("Set-Cookie"))
+	assertStatus(t, response, http.StatusCreated)
+	goassert.True(t, response.Header().Get("Set-Cookie") != "")
```

Comment:
A couple of things here:

1. As discussed the other day, we should try to switch any old `goassert` usages to the `testify/assert` package when we're modifying tests. This is generally a more flexible and better supported library.
2. We should try to switch from `assert.True` to `assert.Equal` or similar, as it gives clearer output when the test fails. This reduces the need for the `log.Printf` lines printing actual values all over the place. Order of parameters makes a difference in these assertion functions, so you'll need to [look up the function signature to verify](https://godoc.org/github.com/stretchr/testify/assert#Equal).
3. Let's also verify that the session ID returned back in the `Set-Cookie` header is the one we inserted into the bucket, instead of "not empty". Maybe something like this:

```suggestion
	assert.Contains(t, response.Header().Get("Set-Cookie"), auth.DefaultCookieName+"="+fakeSession.ID)
```

Reply:
Code changes has been committed.